src/_guides/language/effective-dart/usage.md

```diff
@@ -1013,6 +1013,37 @@ View(Style style, List children)
 an error is reported.
 </div>
 
+
+### DON'T use `new`.
+
+Dart 2 makes the `new` keyword optional. Even in Dart 1, its meaning was never
+clear because factory constructors mean a `new` invocation may still not
+actually return a new object.
+
+The language still permits `new` in order to make migration less painful, but
+consider it deprecated and remove it from your code.
+
+
+### DON'T use `const` when not needed.
+
+In contexts where only constant expressions are allowed, the `const` keyword is
+implicit, doesn't need to be written, and shouldn't. Those contexts are:
+
+* An element expression in a const collection literal.
+* An argument expression in a const constructor call or metadata annotation.
+* The initializer for a const variable declaration.
+* A switch case expression&mdash;the part right after `case` before the `:`, not
+  the body of the case.
+
+(Default values are *not* included in this list because future versions of Dart
+may support non-const default values. So you still need to explicit write
+`const` on the outermost expression there so that it doesn't become a breaking
+change to later support non-const expressions in that context.)
+
+Basically, any place where it would be a static error to write `new` instead of
```

Comment:
I'm not sure what "static error" means. Do you mean where the analyzer would tell you it's an error?

Reply:
Yes. I use that term in a couple of other places. "Compile error" would sort of work too, but I feel that's too focused on a specific action (compilation) when really what we mean is "errors that can be detected without having to run the code".